### PR TITLE
Fix cross: appstream, wlroots, wlogout

### PR DESCRIPTION
**Pull request by Mindavi**
## Description of changes

Some small cross-compilation fixes. Hopefully doesn´t rebuild the world, but I'll retarget if needed.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mindavi** [comment]
@ofborg eval

File: pkgs/development/libraries/appstream/default.nix [hunk 1]
```diff
@@ -79,6 +80,8 @@ stdenv.mkDerivation (finalAttrs: {
     vala
     gperf
   ] ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+    # TODO(Mindavi): Probably should be done for _all_ cross cases?
```
Comment (Aleksanaa) [inline pkgs/development/libraries/appstream/default.nix:83]: Why is appstream related here?

File: pkgs/by-name/wl/wlogout/package.nix [hunk 2]
```diff
@@ -64,6 +64,16 @@ stdenv.mkDerivation (finalAttrs: {
...
+
+    # aarch64 build (cross-compilation) fails with -Werror like this:
+    # aarch64-unknown-linux-gnu-gcc ... -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c11 -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -MD -MQ wlogout.p/main.c.o -MF wlogout.p/main.c.o.d -o wlogout.p/main.c.o -c ../main.c
+    # ../main.c: In function 'set_fullscreen':
+    # ../main.c:499:17: error: unused variable 'mon' [-Werror=unused-variable]
+    # 499 |     GdkMonitor *mon = gdk_display_get_monitor(gdk_display_get_default(), monitor);
+    #     |                 ^~~
+    # cc1: all warnings being treated as errors
+    substituteInPlace meson.build \
+      --replace-warn "werror=true" "werror=false"
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wlogout/package.nix:76]: Maybe `env.NIX_CFLAGS_COMPILE= lib.optionalString <some cross situations> "-Werror=unused-variable"`?

File: pkgs/development/libraries/wlroots/default.nix [hunk 3]
```diff
@@ -127,9 +127,11 @@ rec {
   wlroots_0_17 = generic {
     version = "0.17.2";
     hash = "sha256-Of9qykyVnBURc5A2pvCMm7sLbnuuG7OPWLxodQLN2Xg=";
+    extraNativeBuildInputs = [
+      hwdata
```
Comment (wineee) [inline pkgs/development/libraries/wlroots/default.nix:131]: Why move hwdata to nativeBuildInputs?

**Review by wineee — CHANGES_REQUESTED** [review]

**Comment by danielbarter** [comment]
Just tested, and all these builds are working

File: pkgs/by-name/wl/wlogout/package.nix (hunk 2 above)
Comment (Mindavi) [reply pkgs/by-name/wl/wlogout/package.nix:76]: Thanks for finding that :)

File: pkgs/development/libraries/wlroots/default.nix (hunk 3 above)
Comment (wineee) [reply pkgs/development/libraries/wlroots/default.nix:131]: There are some related discussions: https://lists.openembedded.org/g/openembedded-devel/topic/94909525

**Comment by uninsane** [comment]
these are all building on master (`nix-build -A pkgsCross.aarch64-multiplatform.appstream -A pkgsCross.aarch64-multiplatform.wlroots -A pkgsCross.aarch64-multiplatform.wlogout`): closing

**Comment by uninsane** [comment]
and thanks for the patches. even if master was fixed by some other route, you know i cherry picked these commits in the meantime :)

**Pull request closed without merging**